ui/app/AppLayouts/Profile/views/CommunitiesView.qml

```diff
@@ -90,52 +89,136 @@ SettingsContentBase {
+                    expression: model.joined && model.memberRole !== ownerRole && model.memberRole !== adminRole
                 }
+            }
 
-                onSetCommunityMutedClicked: {
-                    root.profileSectionStore.communitiesProfileModule.setCommunityMuted(communityId, muted)
-                }
+            Heading {
+                text: qsTr("Pending Requests")
+                visible: panelPendingRequests.count
+            }
 
-                onSetActiveCommunityClicked: {
-                    rootStore.setActiveCommunity(communityId)
+            Panel {
+                id: panelPendingRequests
+                filters: ValueFilter {
+                    roleName: "spectated"
+                    value: true
                 }
+            }
+        }
+    }
 
-                onInviteFriends: {
-                    Global.openInviteFriendsToCommunityPopup(communityData,
-                                                             root.profileSectionStore.communitiesProfileModule,
-                                                             null)
-                }
+    component Heading: StatusBaseText {
+        anchors.left: parent.left
+        anchors.leftMargin: Style.current.padding
+        color: Theme.palette.baseColor1
+    }
+
+    component Panel: CommunitiesListPanel {
+        id: panel
+
+        property var filters
+
+        width: parent.width
+
+        model: SortFilterProxyModel {
+            sourceModel: root.profileSectionStore.communitiesList
+            filters: panel.filters
+        }
+
+        onCloseCommunityClicked: {
+            root.profileSectionStore.communitiesProfileModule.leaveCommunity(communityId)
+        }
+
+        onLeaveCommunityClicked: {
+            Global.leaveCommunityRequested(community, communityId, outroMessage)
+        }
+
+        onSetCommunityMutedClicked: {
+            root.profileSectionStore.communitiesProfileModule.setCommunityMuted(communityId, muted)
+        }
+
+        onSetActiveCommunityClicked: {
+            rootStore.setActiveCommunity(communityId)
+        }
+
+        onInviteFriends: {
+            Global.openInviteFriendsToCommunityPopup(communityData,
+                                                     root.profileSectionStore.communitiesProfileModule,
+                                                     null)
+        }
+        onShowCommunityIntroDialog: {
+            Global.openPopup(communityIntroDialogPopup, {
+                communityId: communityId,
+                isInvitationPending: root.rootStore.isCommunityRequestPending(communityId),
+                name: name,
+                introMessage: introMessage,
+                imageSrc: imageSrc,
+                accessType: accessType
+            })
+        }
+    }
+
+    readonly property var communityIntroDialogPopup: Component {
+        id: communityIntroDialogPopup
+        CommunityIntroDialog {
```

Comment:
This `CommunityIntroDialog` should probably get moved to the shared `Popups.qml`